.github/workflows/docker_runner-conda.yml

```diff
@@ -7,52 +7,72 @@ on:
       - main
       - '*staging'
     paths:
+      - '.github/workflows/docker_runner-conda.yml'
       - 'runners/conda-dockerfile'
       - 'runners/r-environment.yml'
       - 'runners/python-environment.yml'
       - 'runners/.condarc'
 
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}/runner-conda
+
 jobs:
   runner-conda:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
     steps:
-    - name: Check out the repo
-      uses: actions/checkout@v3
-
-    - name: Log in to Docker Hub
-      uses: docker/login-action@v2
-      with:
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
-
-    - name: Extract metadata (tags, labels) for Docker
-      if: github.ref == 'refs/heads/main'
-      id: metadata-main
-      uses: docker/metadata-action@v4
-      with:
-        images: geobon/bon-in-a-box
-        tags: |
-          # Image used in compose file
-          type=raw,value=runner-conda
-          # In case we want to replicate a specific legacy setup
-          type=sha,prefix=runner-conda-
-
-    - name: Extract metadata (tags, labels) for Docker, staging branch
-      if: github.ref != 'refs/heads/main'
-      id: metadata-staging
-      uses: docker/metadata-action@v4
-      with:
-        images: geobon/bon-in-a-box
-        tags: |
-          # Image used in compose file
-          type=raw,value=runner-conda-${{github.ref_name}}
-
-    - name: Build and push Docker image
-      uses: docker/build-push-action@v3
-      with:
-        context: runners
-        file: runners/conda-dockerfile
-        push: true
-        tags: ${{ steps.metadata-main.outputs.tags || steps.metadata-staging.outputs.tags }}
-        labels: ${{ steps.metadata-main.outputs.labels || steps.metadata-staging.outputs.labels }}
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker, main branch
+        if: github.ref == 'refs/heads/main'
+        id: metadata-main
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}
+          tags: |
+            # Image used in compose file
+            type=raw,value=latest
+            # In case we want to replicate a specific legacy setup
+            type=sha,value=${{ github.sha }}
```

Comment:
we'd like to know this at runtime, so we can eventually reproduce an environment

Reply:
Won't do: this can be deduced from the label with the full length sha tag.